src/test/java/org/pac4j/javalin/example/JavalinPac4jExample.java

```diff
@@ -144,19 +144,23 @@ private static void protectedPage(Context ctx) {
     }
 
     private static List<CommonProfile> getProfiles(Context ctx) {
-        return new ProfileManager(new Pac4jContext(ctx)).getAll(true);
+        return new ProfileManager<CommonProfile>(new JavalinWebContext(ctx)).getAll(true);
     }
 
     private static void forceLogin(Context ctx, Config config) {
-        Pac4jContext context = new Pac4jContext(ctx);
-        String clientName = context.getRequestParameter("FormClient");
-        Client client = config.getClients().findClient(clientName);
+        JavalinWebContext context = new JavalinWebContext(ctx);
+        String clientName = context.getRequestParameter("FormClient").orElseThrow(() -> {
+            throw new IllegalStateException("Client name not found");
+        });
+        Client client = config.getClients().findClient(clientName).orElseThrow(() -> {
+            throw new IllegalStateException("Client not found");
+        });
         HttpAction action;
         try {
-            action = client.redirect(context);
+            action = (HttpAction) client.getRedirectionAction(context).get();
         } catch (HttpAction e) {
             action = e;
         }
-        config.getHttpActionAdapter().adapt(action.getCode(), context);
+        config.getHttpActionAdapter().adapt(action, context);
```

Comment:
You should use `JavalinHttpActionAdapter.INSTANCE` instead of `config.getHttpActionAdapter()`.